vulcanai2/models/basenetwork.py

```diff
@@ -140,10 +134,35 @@ def __init__(self, name, config, in_dim=None, save_path=None,
         # Compute self.out_dim of the network
         self.out_dim = self._get_out_dim()
 
+    def add_input_network(self, in_network):
+        """
+        Add a new network to  an input for this network.
+
+        New input network will exist at the end of the current set of
+        input_networks which will need to correspond with .
+
+        Parameters
+        ----------
+        in_network : BaseNetwork
+            A network to append to the set of self.input_networks.
+
+        Returns
+        -------
+        None
+
+        """
+        if self.input_networks is None:
+            self.input_networks = nn.ModuleDict()
+        assert isinstance(in_network, BaseNetwork)
+        assert in_network is not self
+        assert isinstance(self.input_networks, nn.ModuleDict)
+        self.input_networks[in_network.name] = in_network
+        self.in_dim = self._get_in_dim()
+
```

Comment:
Just a question - will this affect any GPU/CPU stuff?

Reply:
the input_network function I mean. The fact that you can now add in input networks.

Reply:
that's something we'll need to explore in the GPU PR. this is just implementing it the same way. It may get interesting if your input networks are distributed across different GPU's but as long as the user specifies the right location it shouldn't be that much of a problem. @priyatharsan thoughts?